python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -548,7 +618,23 @@ def _prepare_for_put(self, inMemoryDataset: Any, ref: DatasetRef) -> Tuple[Locat
             The associated `DatasetType` is not handled by this datastore.
         """
         self._validate_put_parameters(inMemoryDataset, ref)
+        return self._determine_put_formatter_location(ref)
 
+    def _determine_put_formatter_location(self, ref: DatasetRef) -> Tuple[Location, Formatter]:
```

Comment:
Differences from `predictLocation()` (now deleted) include passing the dataId to `getFormatter()` and not trapping exceptions from `makeUpdatedLocation()`.  Are you sure that this can be used everywhere the other was being used?

Reply:
@TallJimbo was putting that dataId in on #476 and it was never really used.

I did notice that exception trap and I think that only triggers if a non-file Formatter is somehow used. The `butler.put()` code assumes that can never happen and I'm pretty sure the URI prediction code shouldn't trigger it.